### DISCUSSION TITLE
iOS Wallpaper functionality "motion not available"

### DISCUSSION THREAD

**Discussion by gfsiciliani** [RhetTbull/makelive#21] [Q&A]
Wondering if anyone has had success in producing `.pvt` files that are accepted by iOS's live wallpaper function. Currently, iOS shows _"Motion Not Available - Motion from this Live Photo is not supported as a wallpaper."_

The `.pvt` files I have function flawlessly in Apple Photos and Finder. I have ensured that the video file is 2 seconds in duration, experimented with both 15fps and 30fps, tried both `.mp4` and `.mov` for video formats, ensured both  source files are the same resolution.

**Comment by RhetTbull** [comment]
I don't use motion wallpaper and haven't tried it but there is a `makelive --pvt` option to create a .pvt file. Have you tried that?

**Comment by gfsiciliani** [comment]
Thanks for your quick reply! Yes, I am using the `--pvt` option. Since my initial comment I tried a 12:16 aspect ratio instead of 9:16, as well as 3 second duration with no change in results.

As an artist and developer, I would be happy to take any pointers on resources to learn about whatever check iOS is doing to qualify which live photos are allowed as live wallpapers. There are some mediocre apps in the AppStore (VideoToLive) that pass this qualification given same source file but of course they're not FOSS :)

**Reply by RhetTbull** [reply]
If you can send me a .pvt that does work I can take a look. osxphotos.py@gmail.com

**Reply by gfsiciliani** [reply]
I'm afraid the only files functional for live wallpapers are stuck in the Photos app exportable only as `.heic`+`.mov` files. 
Would it be helpful for me to send working examples even if they're not `.pvt`?

At a base level, I'm curious what meaningful differences exist between a `.pvt` made by makelive and a livephoto in the photos app imported from a 3rd party iOS app.

**Reply by gfsiciliani** [reply]
tried a different approach and got a working `.pvt`, just not with my content yet. I'll send it to your email with more detail momentarily.

**Reply by lofihippo** [reply]
@gfsiciliani what approach did you use to get it working btw? I've tried .jpg + .mov combos with a few files both in .pvt and regular and the generated file doesn't work on iOS photos. It does work on Photos on Mac though.

**Reply by gfsiciliani** [reply]
## Further troubleshooting: 
**attn: @RhetTbull**

Below is the breakdown which believe narrows the focus to the movie file; row 2 in particular being most insightful. For these examples, I took a live photo with known-working live-wallpaper functionality from a 3rd party app and exported the unmodified originals from Photos on macOS. I chose a 3rd party app for this since it seems to replicate my own goal/process of using graphic art instead of iOS-produced video as the source for live-walpapers.

| pvt name | source image | source movie |  live-wallpaper functionality |
|----------|----------|----------|----------|
| `unmodified_original.pvt`  | `unmodified_original.heic`   | `unmodified_original.mov`   | ✅ |
| `newIMG_oldMOV.pvt`        | NEW extracted frame - `heic` | `unmodified_original.mov`   | ✅ |
| `newIMG_newMOV.pvt`        | NEW extracted frame - `heic` | NEW exported movie - `.mp4` | 🚫 |
| `newIMG_newMOV_2.pvt`      | NEW extracted frame - `heic` | NEW exported movie - `.mov` | 🚫 |

_all "NEW" above are derived from `unmodified_original.mov` using kdenlive, Preview and QuickTime for .heic and .mov conversions_

In the zip below are all the pvt files mentioned above as well as the `unmodified_original` .heic and .mov files.
I have not yet looked into the EXIF differences.

[pvt_examples.zip](https://github.com/user-attachments/files/16830274/pvt_examples.zip)

----
@lofihippo It sounds like you and I may have different questions and end goals. "Working" for me means I was able to use makelive to produce a pvt with **live-wallpaper functionality** using `makelive --pvt filename.heic filename.mov` and the source files explained above.

**Reply by lofihippo** [reply]
Thanks for your write up! You're write, i misread your first post. The generated .pvt file works flawlessly on my mac photos and finder app too, same issue with iOS not recognizing the motion

**Reply by fcatapano** [reply]
Hey @gfsiciliani, thanks for the testing and sharing your findings. 

I'm stuck in the process; something needs to be fixed. 

1. I managed to create a `.pvt` using makelive with `live-wallpaper.jpg` and `live-wallpaper.mov` (6:19, video length is less than 3s, h264 codec)
2. The `.pvt` works perfectly in iPhoto and Preview App on OSx
3. I tried to Airdrop the `.pvt` on my iPhone, and it was saved as a file inside Files. 
4. I can not export as image or/and use as live-wallpaper

This is what I tried: 
1.  Converting in `.heic` the image works the video doesn't (the filesize is small compared to `.pvt`) 
2. Tried to share it from iPhoto. 

I'd like to know if someone created a live-wallpaper and shared the usable file. Thanks

**Reply by RhetTbull** [reply]
I'm willing to a take a look at this if someone can get me some samples from other apps that do this correctly. In addition to the fps and aspect ratio and length, there may be other metadata that's required. I don't use motion wallpapers so it's not a problem I've run into.

**Reply by gfsiciliani** [reply]
@RhetTbull - [working_PVTs.zip](https://github.com/user-attachments/files/17350891/working_PVTs.zip)
here is a zip of confirmed functional live-wallpaper PVT files (made with `makelive --pvt`) along with the unmodified originals used to create them.

@fcatapano AirDrop is functional for me in the examples attached above. On the rare occasion, I also ran into AirDrop landing in iOS Files App, but was not able to deduce why. (also, welcome to the thread! big fan of your artwork ^_^)

**Reply by fcatapano** [reply]
Hey @RhetTbull, that would be awesome.

I found and tried out an app called "into - Live Wallpapers." This app allows me to create a live wallpaper from a video and save it to my Camera Roll. However, when I try to share the live photo, my iPhone automatically converts the file into a static `.heic` format.

As @gfsiciliani pointed out, I can share the `.pvt` file via AirDrop, and it works as a live photo and live wallpaper. Unfortunately, I can't share the `.pvt` file in any other way.

Ideally, I want to create a working live wallpaper that can be shared via Google Drive or a similar service. The end user should be able to download the image to their camera roll and easily use it as a live wallpaper.

Thanks a lot for your help @RhetTbull
And thanks @gfsiciliani I'll dm you 👍

**Reply by fcatapano** [reply]
Maybe this might be useful for understanding what's happening: 

1. I upload the `.pvt` file on Dropbox
2. Dropbox sees the `.pvt` as a folder with three files inside `.mp4`, `.jpg` and `.plist`

Can the `.pvt` be converted into an animated `.heic`?

<img width="434" alt="Screenshot 2024-10-15 at 17 17 01" src="https://github.com/user-attachments/assets/5b2274ac-ef29-45af-84b8-53aabcdc4dd4">

**Reply by gfsiciliani** [reply]
@fcatapano Thx for the update and clarification of the issue you're experiencing!

The roadblock that inspired this discussion with occurs at the point of setting the '.pvt' as a wallpaper with motion 
Screenshot below is from the wallpaper setup screen where I experience the roadblock.

Were you able to get past this part? If so, would you be willing to share your .pvt here so I can peek at the exif/metadata?

![motion_not_available](https://github.com/user-attachments/assets/94d8092a-d8e8-4dd7-9df6-a45b286f9835)

**Reply by RhetTbull** [reply]
@gfsiciliani thanks for the sample files. I still haven't figured out what the secret is. Even on my phone, I used two different live photos shot on the iPhone 15 Pro recently and one worked as motion wallpaper and the other did not. I compared the metadata of the two images and was not able to determine why one worked and the other didn't. I will keep exploring as time permits. Ideally, we have a solution where makelive can always adjust the images to work but it would be good to even have a check feature which would at least tell you that a given image/video pair won't work for wallpaper.

**Reply by fcatapano** [reply]
Thanks a lot for your time, @RhetTbull. We own you one! I'll try to share my working test.
